server/executor/tracepollerworker/evaluator_worker.go

```diff
@@ -142,6 +142,11 @@ func (w *tracePollerEvaluatorWorker) ProcessItem(ctx context.Context, job execut
 	log.Printf("[TracePoller] Test %s Run %d: Done polling. (%s)", job.Test.ID, job.Run.ID, reason)
 
 	log.Printf("[TracePoller] Test %s Run %d: Start Sorting", job.Test.ID, job.Run.ID)
+	if job.Run.Trace == nil {
+		newTrace := traces.NewTrace(job.Run.TraceID.String(), []traces.Span{})
+		job.Run.Trace = &newTrace
+	}
```

Comment:
There was a problem caused by this change:
https://github.com/kubeshop/tracetest/pull/3170/files#diff-8477c1b1ce0272a42d729b310d1dacfd4452b274c8de5e0737d701593965b150R84

Which make every incoming trace without spans to be marked as nil, but for no tracing mode, this is expected